scrapper.py

```diff
@@ -25,39 +39,136 @@ class Crawler:
         """
         Returns seed_urls param
         """
-        pass
+        return self.urls
 
 
-def prepare_environment(base_path):
+class HTMLParser:
     """
-    Creates ASSETS_PATH folder if not created and removes existing folder
+    Parser implementation
     """
-    pass
 
+    def __init__(self, article_url, article_id):
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(self.article_url, self.article_id)
 
-def validate_config(crawler_path):
-    """
-    Validates given config
-    """
-    pass
+    def parse(self):
+        """
+        Extracts all necessary data from the article web page
+        """
+        response = requests.get(self.article_url, headers=HEADERS, allow_redirects=True)
+
+        article_bs = BeautifulSoup(response.text, 'lxml')
+
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+
+        return self.article
+
+    def _fill_article_with_text(self, article_bs):
+        """
+        Fills the Article instance with text
+        """
+        text_bs = article_bs.find(class_="text-article").text
+        self.article.text = text_bs.text
+
+    def _fill_article_with_meta_information(self, article_bs):
+        """
+        Fills the Article instance with meta information
+        """
+        title_bs = article_bs.find(class_='top_big_img_article__info__inside__title').text
+        self.article.title = title_bs
+        print(title_bs)
+
+        date_bs = article_bs.find(class_="top_big_img_article__info__inside__time").text
+        date = datetime.strptime(date_bs, "%d.%m.%Y, %H:%M")
+        self.article.date = date
+
+        tags_bs = article_bs.find('div', class_='hash_tags')
+        self.article.tags = tags_bs
+
+        try:
+            author_bs = article_bs.find(class_='top_big_img_article__info__inside__author').text
+        except AttributeError:
+            author_bs = 'NOT FOUND'
+        self.article.author = author_bs
+
+    def prepare_environment(base_path):
+        """
+        Creates ASSETS_PATH folder if not created and removes existing folder
+        """
+        path = Path(base_path)
+        if path.exists():
+            shutil.rmtree(base_path)
+        base_path.mkdir(exist_ok=True, parents=True)
+
+    def validate_config(crawler_path):
+        """
+        Validates given config
+        """
+        with open(crawler_path) as file:
+            config = json.load(file)  # load is for decoding
+
+        seed_urls = config['seed_urls']
+        max_articles = config['total_articles_to_find_and_parse']
+
+        if not seed_urls:
+            raise IncorrectURLError
+
+        for url in seed_urls:
+            if not re.match(r'https?://', url):
+                raise IncorrectURLError
+
+        if not isinstance(max_articles, int) or max_articles <= 0:
+            raise IncorrectNumberOfArticlesError
+
+        if max_articles > 100:
```

Comment:
Take a slightly bigger number of articles as 100 is a required one, and it is odd if we forbid collecting 101, for example